### PR TITLE
feat: add native addon to cidentity API

### PR DESCRIPTION
**Pull request by steff456**
Part of https://github.com/stdlib-js/todo/issues/1454

## Description

> What is the purpose of this pull request?

This pull request:

- Adds support for Complex128 dtype in the C interface  
- Adds native addon with benchmark, tests, examples and readme update

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Part of https://github.com/stdlib-js/todo/issues/1454

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/cidentity/README.md [hunk 1]
```diff
@@ -132,18 +132,27 @@ for ( i = 0; i < 100; i++ ) {
...
 ```c
-#include <complex.h>
+#include "stdlib/complex/float64.h"
+#include "stdlib/complex/real.h"
+#include "stdlib/complex/imag.h"
 
-double complex y = stdlib_base_cidentity( 2.0+2.0*I );
-// returns 2.0+2.0*I
+stdlib_complex128_t z = stdlib_complex128( 2.5, -1.5 );
+
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/README.md]:
```suggestion
```

File: lib/node_modules/@stdlib/math/base/special/cidentity/benchmark/c/Makefile
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/benchmark/c/Makefile:1]: We can just remove this file.

File: lib/node_modules/@stdlib/math/base/special/cidentity/benchmark/c/benchmark.c
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/benchmark/c/benchmark.c:1]: We can remove this file. The benchmark is unlikely to be informative due to compiler optimizations.

File: lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+var Complex128 = require( '@stdlib/complex/float64' );
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Evaluates the identity function for a double-precision complex floating-point number.
+*
+* @param {Complex128} z - input value
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js:33]:
```suggestion
* @private
* @param {Complex128} z - input value
```

File: lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+* var imag = require( '@stdlib/complex/imag' );
+*
+* var v = cidentity( new Complex128( -1.0, 2.0 ) );
+* // throws <ReferenceError>
+*
+* var re = real( v );
+* // throws <TypeError>
+*
+* var im = imag( v );
+* // throws <TypeError>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js]: The return values are incorrect. Likely need to ensure that the add-on is compiled when committing, due to lint auto-fixing. 

cc @Planeshifter

File: lib/node_modules/@stdlib/math/base/special/cidentity/manifest.json [hunk 4]
```diff
@@ -24,15 +26,51 @@
...
+		  ]
+		},
+		{
+		  "task": "benchmark",
+		  "src": [
+			"./src/main.c"
+		  ],
+		  "include": [
+			"./include"
+		  ],
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/manifest.json]: This file has mixed TAB/SPACE indentation. Provided EditorConfig is setup, this should be automatically handled.

File: lib/node_modules/@stdlib/math/base/special/cidentity/src/main.c [hunk 5]
```diff
@@ -17,7 +17,6 @@
 */
 
 #include "stdlib/math/base/special/cidentity.h"
-#include <complex.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/src/main.c:21]:
```suggestion
#include "stdlib/complex/float64.h"

```

Missing the header for `stdlib_complex128_t`.

File: lib/node_modules/@stdlib/math/base/special/cidentity/src/main.c [hunk 6]
```diff
@@ -26,8 +26,21 @@
 * @return        input value
 *
 * @example
-* double complex y = stdlib_base_cidentity( 3.0+3.0*I );
+* #include "stdlib/complex/float64.h"
+* #include "stdlib/complex/real.h"
+* #include "stdlib/complex/imag.h"
+*
+* stdlib_complex128_t z = stdlib_complex128( 3.0, 3.0 );
+*
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/src/main.c]:
```suggestion
```

**Review by kgryte — CHANGES_REQUESTED** [review]
Thanks, @steff456! Three changes should be made before merge:

1. We can remove the "vanilla" C benchmarks and keep the `benchmark/native` benchmarks. The vanilla benchmarks are not likely to be informative.
2. Need to fix return annotations in the JSDoc comments.
3. Need to fix indentation in the `manifest.json` file.

Once resolved, this PR should be ready to merge.

File: lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js (hunk 3 above)
Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js]: @steff456 These return values do not seemed to have been fixed.

File: lib/node_modules/@stdlib/math/base/special/cidentity/manifest.json [hunk 7]
```diff
@@ -24,15 +26,51 @@
...
-			"src": [
-				"./src/main.c"
-			],
-			"include": [
-				"./include"
-			],
-			"libraries": [],
-			"libpath": [],
-			"dependencies": []
+		  "task": "build",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/manifest.json]: @steff456 The indentation in this file is mixed tabs and spaces. Should be normalized according to EditorConfig conventions.

**Review by kgryte — CHANGES_REQUESTED** [review]
@steff456 Thanks for removing the obsolete benchmark files. Looks like we still need to resolve the indentation and JSDoc issues before this PR can be merged.

File: lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js (hunk 3 above)
Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js]: I'm not sure what is happening, I change them and when I go ahead and do a commit with the `make commit` command it changes to these values... 

I have the addon already compiled 😕

Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js]: Update: I created the commit with the `--no-verify` flag, and this is the only PR where I had this issue. I'll check in future PRs if the behavior repeats.

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/cidentity/lib/native.js]: Once merged, I'll pull down and try to compile on my end.

File: lib/node_modules/@stdlib/math/base/special/cidentity/examples/c/example.c [hunk 8]
```diff
@@ -17,19 +17,31 @@
 */
 
 #include "stdlib/math/base/special/cidentity.h"
+#include "stdlib/complex/float64.h"
+#include "stdlib/complex/reim.h"
 #include <stdio.h>
-#include <complex.h>
 
 int main( void ) {
```
Comment (steff456) [inline lib/node_modules/@stdlib/math/base/special/cidentity/examples/c/example.c:24]: @kgryte, I'm not sure if we want to leave the `void` here. Can you please confirm this before merging?

Comment (kgryte) [reply lib/node_modules/@stdlib/math/base/special/cidentity/examples/c/example.c:24]: @steff456 We do. In C, this is how one communicates that a function accepts zero-arguments. We had previously been inconsistent with including `void` (e.g., examples tended not to include, but benchmarks did include). Not including it is deprecated behavior. Hence, its current inclusion and best practice going forward.

Comment (steff456) [reply lib/node_modules/@stdlib/math/base/special/cidentity/examples/c/example.c:24]: Great! I'll take it into account for the open PRs and the ones coming 😊

File: lib/node_modules/@stdlib/math/base/special/cidentity/manifest.json [hunk 9]
```diff
@@ -1,38 +1,76 @@
 {
-	"options": {},
-	"fields": [
-		{
-			"field": "src",
-			"resolve": true,
-			"relative": true
+		"options": {
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/cidentity/manifest.json]: Need to fix spacing in this file.

**Review by kgryte — CHANGES_REQUESTED** [review]
Need to fix spacing in manifest.json.

**Comment by kgryte** [comment]
@steff456 Would you mind creating an internal todo for a `manifest.json` linter?

**Review by kgryte — APPROVED** [review]
Thanks, @steff456!

**Pull request merged**